### PR TITLE
fix: update auth logs filtering to use metadata keys.

### PR DESCRIPTION
**Pull request by Ziinc**
Updates filter generation and auth logs ui to use log metadata instead of event_message

- [x] verify auth logs working
- [x] Verify ui filters is working

https://github.com/supabase/supabase/assets/22714384/b1118a04-f7ba-42e6-b391-1a82c1c93d64

### CODE REVIEW

File: studio/components/interfaces/Settings/Logs/Logs.utils.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ const getDotKeys = (obj: { [k: string]: unknown }, parent?: string): string[] =>
  *
  * @returns a where statement with WHERE clause.
  */
-const _genWhereStatement = (table: LogsTableName, filters: Filters) => {
+export const _genWhereStatement = (table: LogsTableName, filters: Filters) => {
```
Comment (alaister) [inline studio/components/interfaces/Settings/Logs/Logs.utils.ts:68]: small nit: now that this is exported, its name shouldn't start with an underscore

**Review by alaister — CHANGES_REQUESTED** [review]
Looks good!

There seems to be a failing test, though – would you mind fixing it? 🙏🏻

File: studio/components/interfaces/Settings/Logs/Logs.utils.ts (hunk 1 above)
Comment (Ziinc) [reply studio/components/interfaces/Settings/Logs/Logs.utils.ts:68]: i've removed the export, we don't need to export it.

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**